Makefile: wait for swtmp sock to be ready

hallyn opened the pull request:
Also clean up from previous swtpm runs.

Signed-off-by: Serge Hallyn <serge@hallyn.com>

hallyn committed:
Makefile: wait for swtmp sock to be ready

Also clean up from previous swtpm runs.

Signed-off-by: Serge Hallyn <serge@hallyn.com>

smoser commented on Makefile:
```diff
@@ -72,8 +72,12 @@ qemu-esp: ovmf_fw.fd ovmf_vars.fd
+	killall swtpm
 	(swtpm socket --tpmstate dir=.tpm2 --tpm2 \
 		--ctrl type=unixio,path=.tpm2/swtpm-sock &)
+	(while [ ! -e .tpm2/swtpm-sock ]; do sleep 1; done)
```
parens are superfluous.

smoser commented on Makefile:
```diff
@@ -72,8 +72,12 @@ qemu-esp: ovmf_fw.fd ovmf_vars.fd
 	# NOTE: swtpm normally exits when qemu exits, no need to cleanup
 	[ -d .tpm2 ] || mkdir .tpm2
+	rm -f .tpm2/swtpm-sock
+	killall swtpm
```
i'd like that redirected to /dev/null so we don't have "no swtpm around" errors.... and won't `killall swtpm` exit non-zero (and fail make) if there is no 'swtpm' ?  And *then* we don't really want to kill *all* swtpm, just this one.

```shell
$ killall swtpm; echo $?
swtpm: no process found
1
```

hallyn replied:
> won't killall swtpm exit non-zero (and fail make) if there is no 'swtpm'

Maybe.  I must have added that while pushing the patch because the tree I use does not have that killall.

(This is why I don't like making tons of stylistic changes as part of review. The final version won't be nearly as well tested).

So I'll drop that, thanks (even though || true would of course fix that specific problem)

smoser reviewed (changes requested):
i think the easiest way to do this is to put the tpm starting code into a './tools/run-sw-tpm <dir>' program and have it use '--pid=' so then it can kill the previous swtpm by pid if there is one.

The one thing i'm confused by though, is why we dpn't see a race this in atomix-vm-builder's --tpm.  There, we just execute swtpm and immediately start qemu.

hallyn commented:
> The one thing i'm confused by though, is why we dpn't see a race this in atomix-vm-builder's --tpm. There, we just execute swtpm and immediately start qemu.

It's conceivable that that golang code is doing enough cruft between starting the swtpm and starting qemu that the race just doesn't show itself.

hallyn committed:
create tools/run/sw-tpm to run the swtpm

Makefile runs swtpm in two separate places already, so
commonize that code as Scott suggested.

Signed-off-by: Serge Hallyn <serge@hallyn.com>
